CardCarousel/CardCarousel.js

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from 'react'
+import Arrow from './Arrow'
+
+export default class CardCarousel extends Component {
+  constructor(props) {
+    super(props)
+    this.state = ({
+      currentPracticeCardIndex: 0,
+      showAnswer: false,
+    })
+    this.hideAnswer = this.hideAnswer.bind(this);
+    this.nextImage = this.nextImage.bind(this);
+    this.previousImage = this.previousImage.bind(this);
+  }
+
+  previousImage() {
+    const lastIndex = this.props.data.length - 1;
+    const { currentPracticeCardIndex } = this.state;
+    const shouldResetIndex = currentPracticeCardIndex === 0;
+    const index = shouldResetIndex ? lastIndex : currentPracticeCardIndex - 1;
+    this.setState({
+      currentPracticeCardIndex: index,
+    });
+  }
+
+  nextImage() {
+    const lastIndex = this.props.data.length - 1;
+    const { currentPracticeCardIndex } = this.state;
+    const shouldResetIndex = currentPracticeCardIndex === lastIndex;
+    const index = shouldResetIndex ? 0 : currentPracticeCardIndex + 1;
+    this.setState({
+      currentPracticeCardIndex: index,
+    });
+  }
+
+  hideAnswer(event) {
+    this.setState({
+      showAnswer: !this.state.showAnswer,
+    })
+  }
+
+  render() {
+    const item = this.props.data[this.state.currentPracticeCardIndex]
+    let showAnswer = "Click to Show Answer!"
+    if (this.state.showAnswer) {
+      showAnswer = (
+        <strong>
+          {item.answer}
+        </strong>
+      )
+    }
+    return (
+      <React.Fragment>
+        <h1 className="practice">
+Practice your flashcards!
+        </h1>
+        <div className="carousel">
+          <Arrow direction="left" clickFunction={this.previousImage} />
```

Comment:
If you are passing a prop that looks like an event handler (for `click`, `submit`, `keypress`, ...etc), use the `on` prefix convention.

```jsx
<Arrow direction="left" onClick={this.previousImage}/>
```

Scan your code for other opportunities to improve prop names.